aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -294,7 +297,12 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
     values_tmp = at::empty_strided(self_.sizes(), self_.strides(), self_.options());
     values_ptr_ = values_tmp.data_ptr();
   } else {
-    values_ptr_ = values.data_ptr();
+    if (newself || get_overlap_status(self, values) == MemOverlapStatus::NO) {
+      values_ptr_ = values.data_ptr();
+    } else {
+      values_tmp = at::empty_like(values);
+      values_ptr_ = values_tmp.data_ptr();
+    }
   }
```

Comment:
Can this be simplified as the following?
```C++
if (values.strides() == self_.strides() && (newself || get_overlap_status(self, values) == MemOverlapStatus::NO)) {
  values_ptr_ = values.data_ptr();
} else {
  values_tmp = at::empty_like(values);
  values_ptr_ = values_tmp.data_ptr();
}
```

Reply:
Should it be `values_tmp = at::empty_like(self_)`, like it's in new line 297 now? And to be clear, you are suggesting replacing lines 296 - 305 with this? That's a good suggestion.

Reply:
What is the difference between `empty_like` and `empty_strided` when the strides passed are from the same tensor?

Reply:
Yes, I am talking about 296 - 305

Reply:
What is the difference between `empty_like` and `empty_strided` when the strides passed are from the same tensor?

Reply:
They are the same I think.